app/helpers/sessions_helper.rb

```diff
@@ -0,0 +1,22 @@
+module SessionsHelper
+
+  # Logs in the given user.
+  def log_in user
+    session[:user_id] = user.id
+  end
+
+  # Returns the current logged-in user (if any).
+  def current_user
+    @current_user ||= User.find_by(id: session[:user_id]) if session[:user_id]
+  end
+
+  # Returns true if the user is logged in, false otherwise.
+  def logged_in?
+    !current_user.nil?
```

Comment:
Dùng `present?` thay vì sử dụng nil và nghịch đảo kết quả

Nói cho chị biết `nil?`, `blank?` và `present?` có gì khác nhau

Reply:
*nil?  cung cấp bởi ruby.
-nil? dùng cho mọi đối tượng trong ruby và trả về true chỉ khi nil.nil?

*blank? cung cấp bởi rails.
-trả về true khi nil và false.
-trong array và hash thì khi nào empty thì là blank?
-còn trong strings blank? còn xác định khoảng trắng trong chuỗi.

*present?
- thì ngược lại với blank?
- check not nil, not false, not empty là kiểm tra tồn tại của object.